pkg/volume/util/util_test.go

```diff
@@ -942,3 +943,249 @@ func TestGetPodVolumeNames(t *testing.T) {
+				Spec: v1.PersistentVolumeSpec{
+					NodeAffinity: &v1.VolumeNodeAffinity{
+						Required: &v1.NodeSelector{
+							NodeSelectorTerms: []v1.NodeSelectorTerm{
+								{
+									MatchExpressions: []v1.NodeSelectorRequirement{
+										{
+											Key:      "foo",
+											Operator: v1.NodeSelectorOpIn,
+										},
+										{
+											Key:      "bar",
+											Operator: v1.NodeSelectorOpIn,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "PV node affinity required single match expression with no values",
+			pv: &v1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Spec: v1.PersistentVolumeSpec{
+					NodeAffinity: &v1.VolumeNodeAffinity{
+						Required: &v1.NodeSelector{
+							NodeSelectorTerms: []v1.NodeSelectorTerm{
+								{
+									MatchExpressions: []v1.NodeSelectorRequirement{
+										{
+											Key:      v1.LabelHostname,
+											Operator: v1.NodeSelectorOpIn,
+											Values:   []string{},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedNodeNames: []string{},
+		},
+		{
+			name: "PV node affinity required single match expression with single node",
+			pv: &v1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Spec: v1.PersistentVolumeSpec{
+					NodeAffinity: &v1.VolumeNodeAffinity{
+						Required: &v1.NodeSelector{
+							NodeSelectorTerms: []v1.NodeSelectorTerm{
+								{
+									MatchExpressions: []v1.NodeSelectorRequirement{
+										{
+											Key:      v1.LabelHostname,
+											Operator: v1.NodeSelectorOpIn,
+											Values: []string{
+												"node1",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedNodeNames: []string{
+				"node1",
+			},
+		},
+		{
+			name: "PV node affinity required single match expression with multiple nodes",
+			pv: &v1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Spec: v1.PersistentVolumeSpec{
+					NodeAffinity: &v1.VolumeNodeAffinity{
+						Required: &v1.NodeSelector{
+							NodeSelectorTerms: []v1.NodeSelectorTerm{
+								{
+									MatchExpressions: []v1.NodeSelectorRequirement{
+										{
+											Key:      v1.LabelHostname,
+											Operator: v1.NodeSelectorOpIn,
+											Values: []string{
+												"node1",
+												"node2",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedNodeNames: []string{
+				"node1",
+				"node2",
+			},
+		},
+		{
+			name: "PV node affinity required multiple match expressions with multiple nodes",
+			pv: &v1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Spec: v1.PersistentVolumeSpec{
+					NodeAffinity: &v1.VolumeNodeAffinity{
+						Required: &v1.NodeSelector{
+							NodeSelectorTerms: []v1.NodeSelectorTerm{
```

Comment:
Construct an additional case for nodeAffinity like:

```
- NodeSelectorTerms[0]
    - MatchExpressions[0]: hostname in [node1, node2]
    - MatchExpressions[0]: hostname in [node2, node3]
- NodeSelectorTerms[1]:
    - MatchExpressions[0]: hostname in [node1]
```

The final result should be [node1, node2]

Reply:
Thanks, added test case described above.